### PR TITLE
feat(grouping): Error code project_not_hierarchical

### PR DESCRIPTION
**Pull request by jjbayer**
In grouping breakdown tab, distinguish between non-hierarchical issues and non-hierarchical projects.

Fix https://getsentry.atlassian.net/browse/INGEST-200

### CODE REVIEW

**Review by priscilawebdev — APPROVED** [review]
lgtm

File: src/sentry/api/endpoints/grouping_levels.py [hunk 1]
```diff
@@ -28,12 +29,18 @@ class MissingFeature(SentryAPIException):
     message = "This project does not have the grouping tree feature."
 
 
-class NotHierarchical(SentryAPIException):
+class IssueNotHierarchical(SentryAPIException):
     status_code = status.HTTP_403_FORBIDDEN
     code = "not_hierarchical"
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/grouping_levels.py:34]: should we update the code to "issue_not_hierarchical" ?

Comment (jjbayer) [reply src/sentry/api/endpoints/grouping_levels.py:34]: Will do.

**Review by priscilawebdev — APPROVED** [review]
tested locally and everything looks good 🚢🇮🇹

**Pull request merged**